python/whylogs/api/store/query.py

```diff
@@ -1,21 +1,36 @@
-from dataclasses import dataclass, field
+from abc import ABC
 from datetime import datetime
 from typing import Optional
 
 
-@dataclass
-class DateQuery:
-    profile_name: str
-    start_date: datetime
-    end_date: Optional[datetime] = field(default=None)
+class BaseQuery(ABC):
+    def __init__(self):
+        self.dataset_id = None
+
```

Comment:
Any long term vision of what these things will actually share? It seems like the base query shouldn't actually have a dataset_id because I can imagine other types of queries that don't care about that. DateQuery doesn't seem like it should care about that, just date ranges. It would be weird to have a base class and then manually specify all of the things the subclasses have in common too.

An alternative is to not have a base class and just export a type alias 

```python
Query = Union[DateQuery, DatasetIdQuery]
```

Reply:
if you have two Projects/Dataset Ids - A and B - persisted to the same store, how do you query A by a date range?

Reply:
btw, made the suggested change to the classes :)

Reply:
You would also have an AND/OR abstraction to enable more combinations of things. So that might look like this:

```python
filter = Or(
    And(DateQuery(...), IdQuery(id1)),
    And(DateQuery(...), IdQuery(id2)),
)
```

Reply:
oh, ok. 
my idea is slightly different, where you'll only need one type of Query object in order to get what you need from the store.
so in your example it would be something like

```python
filter = Or(
    And(DateQuery(date_range, id1),
    And(IdQuery(id2)),
)
```

Reply:
resolving this conversation as this turned out to be a future change, described in: https://github.com/whylabs/whylogs/issues/1003